keyboards/primekb/meridian/meridian.c

```diff
@@ -23,46 +23,42 @@ void keyboard_pre_init_kb(void){
     keyboard_pre_init_user();
 }
 
-//Initialize all RGB indicators to 'off'
+//Refernce the following QMK documentation for implementing Lighting Layers and layer change indication:
+// https://beta.docs.qmk.fm/using-qmk/hardware-features/lighting/feature_rgblight#lighting-layers
+
+// Light the top LED when caps lock is active.
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_WHITE}       // Light 1 LEDs, starting with LED 0
+);
+// Light the middle LED when num lock is active. 
+const rgblight_segment_t PROGMEM my_numlock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {1, 1, HSV_WHITE}       // Light 1 LEDs, starting with LED 1
+);
+// Light the bottom LED when scroll lock is active. 
+const rgblight_segment_t PROGMEM my_scroll_lock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {2, 1, HSV_WHITE}       // Light 1 LEDs, starting with LED 2
+);
+
+// Now define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    my_capslock_layer,
+    my_numlock_layer,
+    my_scroll_lock_layer
+);
+
 __attribute__((weak))
 void keyboard_post_init_user(void) {
-    rgblight_setrgb_at(0, 0, 0, 0); // [..., 0] = top LED
-    rgblight_setrgb_at(0, 0, 0, 1); // [..., 1] = middle LED
-    rgblight_setrgb_at(0, 0, 0, 2); // [..., 2] = bottom LED
+    rgblight_layers = my_rgb_layers; //Enable the LED layers
+    rgblight_disable();    //set RGB lights to off by default. 
 }
 
-//Indicator light function
 bool led_update_kb(led_t led_state) {
     bool res = led_update_user(led_state);
-    if (res) {
- //       writePin(B12, !led_state.caps_lock);  //Un-comment this line to enable in-switch capslock indicator
-    if (led_state.caps_lock) {
-        rgblight_setrgb_at(0, 255, 0, 0); //green
-    } else {
-        rgblight_setrgb_at(0, 0, 0, 0);
+    if(res) {
+        writePin(B12, !led_state.caps_lock);
     }
-    if (led_state.num_lock) {
-        rgblight_setrgb_at(0, 0, 255, 1); //blue
-    } else {
-        rgblight_setrgb_at(0, 0, 0, 1);
-    }
-   if (led_state.scroll_lock) {          
-        rgblight_setrgb_at(255, 0, 0, 2); //red
-    } else {
-        rgblight_setrgb_at(0, 0, 0, 2);
-    }
-}
-    return res;
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    rgblight_set_layer_state(1, led_state.num_lock);
+    rgblight_set_layer_state(2, led_state.scroll_lock);
```

Comment:
Should these be inside of the `if(res) {}` block instead?

Reply:
They should be, IMO, yeah. Otherwise, users can't over ride the feature.